### DISCUSSION TITLE
Byzantine fault-tolerant consensus algorithm?

### DISCUSSION THREAD

**Discussion by 0x0factor** [community/community#124087] [Programming Help]
### Body

 I'm tasked with implementing a Byzantine fault-tolerant consensus algorithm for a permissioned blockchain network with stringent latency requirements. Could you offer guidance on designing an algorithm that balances resilience with low-latency message propagation?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by kcraftcode** [comment]
Designing a Byzantine fault-tolerant consensus algorithm for a permissioned blockchain network requires careful consideration of trade-offs between fault tolerance and performance. One approach is to explore variants of Practical Byzantine Fault Tolerance (PBFT) optimized for low-latency environments, such as HoneyBadger BFT or Tendermint. These algorithms leverage techniques like asynchronous communication and pipelined processing to achieve fast finality while maintaining Byzantine fault tolerance. Let's delve deeper into your specific latency constraints and architectural considerations to devise an optimal solution.

**Comment by ebndev** [comment]
Hi @johnsmith0212 & @kentaurse, thanks for being a part of the GitHub Community!

Please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
